lib/webtask.js

```diff
@@ -708,6 +756,66 @@ exports.ensure_authenticated = function(ctx) {
             });
         }
 
+        function validate_auth_token(auth) {
+            var auth_token;
+            var decoded_auth_token;
+            var userinfo;
+            var aud_should_be;
+            var secret = auth.secret || ctx.config.default_auth0_client_secret.current;
+
+            if(auth.account)
+                aud_should_be = auth.account.client_id;
+            else
+                aud_should_be = req.headers.host + urlParse(req.url).pathname;
+
+            try {
+                secret = new Buffer(secret, 'base64');
+            } catch(e) {
+                throw { code: 403, message: 'JWT_SECRET must be base64 encoded: ' + e.message, req_id: req.req_id };
+            }
+
+            try {
+                auth_token = (req.query.key || req.headers['authorization']
+                    .match(/^bearer\s+(.+)$/i)[1]);
+            } catch(e) {
+                throw { code: 400, message: 'could not parse authorization header' + e.message, req_id: req.req_id };
+            }
+
+            try {
+                userinfo = jsonwebtoken.verify(auth_token, secret);
+            } catch(e) {
+                throw { code: 401, message: 'invalid authorization token signature', req_id: req.req_id };
+            }
+
+            if(!userinfo.aud)
+                throw { code: 403, message: 'no authorization token \'aud\' claim', req_id: req.req_id };
+
+            if(userinfo.aud !== aud_should_be)
+                throw { code: 403, message: 'invalid authorization token \'aud\' claim', req_id: req.req_id };
+
+            if(auth.email) {
```

Comment:
The *authorization* (as opposed to *authentication*) should be possible in userland - in the webtask that triggers lock perhaps, before it invokes the "user" webtask.

In any case, it is probably not a good idea to bake such a prescriptive authorization model at such a low level in the platform. I think it is OK to require that in the low level HTTP API scenarios in which webtasks accept arbitrary JWTs (as opposed those issued by lock), developers are doing authorization on their own in code. @woloski?